### DISCUSSION TITLE
The model created by gmsh has no edges and corners in Unity. Is there any solution?

### DISCUSSION THREAD

**Discussion by MingSongJiang** [InfinyTech3D/SofaUnity#128] [General topics regarding SofaUnity]
Hello,

The model created by gmsh has no edges or corners in Unity. The edges and corners of a rectangular block look rounded. Is there any solution?
![Unity_GCTv8HX8Rm](https://github.com/user-attachments/assets/e48fee18-d09e-4630-ac53-49739026a1de)

Thanks!

**Comment by epernod** [comment]
Hi,

are you sure it is not the material that gives that impression. You can use the `wireframe` material to check the geometry.
Otherwise it might means that the normals are not well exported.

Best regards,

**Comment by CherryJGit** [comment]
Hello，
I'd like to ask if there is no problem of mapping confusion with the model after you import it and apply the material. If there isn't, could you please share the whole process from the obj model to the msh?
Thanks!

**Reply by epernod** [reply]
Hi,

if you are using a tetrahedral mesh (.msh) you need either to create a SOFA barycentric mapping in your scene (.scn) between your obj file with the texture coordinate and the simulated mesh.
Otherwise you can also look at the engine NearestTexcoordsMap used here: https://github.com/InfinyTech3D/InfinyToolkit/blob/main/examples/NearestTexcoordsMap.scn

Best regards,
